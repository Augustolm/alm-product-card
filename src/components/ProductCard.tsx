import styles from '../styles/styles.module.css';
import { useProducts } from '../hooks/useProducts';
import React, { createContext } from 'react';
import {
  ProductContextProps,
  Product,
  onChangeArgs,
  InitialValues,
  ProductCardHandlers,
} from '../interfaces/interfaces';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export interface Props {
  product: Product;
  //children?: ReactElement | ReactElement[];
  children: (args: ProductCardHandlers) => JSX.Element;
  className?: string;
  style?: React.CSSProperties;
  onChange?: (args: onChangeArgs) => void;
  value?: number;
  initialValues?: InitialValues;
}

export const ProductCard = ({
  children,
  product,
  className,
  style,
  onChange,
  value,
  initialValues,
}: Props) => {
  const {
    counter,
    increseBy,
    maxCount,
    isMaxCountReached,
    reset,
  } = useProducts({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider
      value={{
        counter,
        increseBy,
        product,
        maxCount,
      }}
    >
      <div className={`${styles.productCard} ${className} `} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          reset,
          maxCount: initialValues?.maxCount,
          product,
          increseBy,
        })}
      </div>
    </Provider>
  );
};
