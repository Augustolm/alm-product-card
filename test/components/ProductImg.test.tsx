import * as React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/productos';

describe('Product Imagen', () => {
  test('debe de mostrar el componente correectamente con la imagen personalizado', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://www.adslzone.net/app/uploads-adslzone.net/2019/04/borrar-fondo-imagen.jpg" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el componente con la imagen deel producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
