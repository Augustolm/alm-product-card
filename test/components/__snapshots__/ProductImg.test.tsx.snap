// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Imagen debe de mostrar el componente con la imagen deel producto 1`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Imagen"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Product Imagen debe de mostrar el componente correectamente con la imagen personalizado 1`] = `
<img
  alt="Imagen"
  className="productImg undefined"
  src="https://www.adslzone.net/app/uploads-adslzone.net/2019/04/borrar-fondo-imagen.jpg"
/>
`;
